CONTRIBUTING.md

```diff
@@ -54,6 +54,21 @@ $ make changelog
 ````
 …and follow the prompts on screen.
 
+### Pull Request Descriptions
+
+Write clear, informative PR descriptions that help reviewers understand your changes and provide effective feedback. A good PR description serves as documentation for your changes and helps maintainers understand the impact and context.
+
+**Important**: We use squash merge, so your PR description will become the git commit message. Write it accordingly.
+
+The PR description should include:
+
+- **What**: A clear summary of what changes are being made (often the PR title is sufficient)
+- **Why**: The motivation, context, or problem being solved
+- **How**: Optionally a brief explanation of the approach taken (if not obvious). For detailed implementation explanations, consider adding comments in the code instead.
+- **References**: Links to related issues, RFCs, or previous PRs
```

Comment:
@tgummerer Ideally we can keep these in order (what aka title, why, how, refs). We could de-emphasize all the points except make the "why" part bold and explicitly call it out as the most important.

```suggestion
- What: A clear summary of what changes are being made (often the PR title is sufficient)
- **Why: The motivation, context, or problem being solved. This is the most important part of the description.**
- How: Optionally a brief explanation of the approach taken (if not obvious). For detailed implementation explanations, consider adding comments in the code instead.
- References: Links to related issues, RFCs, or previous PRs
```